### ISSUE TITLE
Doorman - Scanner - Auto Check-In

### ISSUE THREAD

**Issue by Mcscott142** [revelrylabs/bn-mobile-react#238]
## Background 
When a Doorman scans a ticket he should have different settings for the Scanner. A Manual Check-in, an Auto Check-in, and a pre-event scan. This is for an Auto Check-In where the Check-in completes without having to confirm by selecting complete checkin

See Video of flow here: https://vimeo.com/298395671

Via @zmelamed  
>So there are 2 checkin "modes", If the Auto-checkin is active it should auto checkin as soon as it's scanned. Show the success state for 3 seconds, and then switch right back into scan mode. If the auto-checkin mode is not active it will require them to click Complete checkin after scanning.

## Scenario: A Doorman checks someone in
- [ ] Given I open the Scanner
- [ ] And Auto Check In is toggled on
- [ ] When I scan a Valid ticket
- [ ] Then I receive a check in complete message
- [ ] And I can scan another ticket

### Mock for Check In Mode
<img width="337" alt="big_neon" src="https://user-images.githubusercontent.com/6877299/47919141-9f04c100-de7c-11e8-9df1-a976310b837c.png">

**Comment by Mcscott142** [comment]
AC Updated, marked Score

**Comment by Mcscott142** [comment]
AC updated. @zmelamed In order to get some sample QR codes/tickets while we work this, guessing we just go to the Web App and purchase some tickets with sample users?

**Comment by brittanygay** [comment]
@chas-mcmahon let me know when this is ready for a design pass!

**Comment by zmelamed** [comment]
> AC updated. @zmelamed In order to get some sample QR codes/tickets while we work this, guessing we just go to the Web App and purchase some tickets with sample users?

Yes I think that'd be the best way to do this. Let me know if you have issues working with the staging web app.

**Comment by brittanygay** [comment]
@chas-mcmahon I just pushed up the styles for the scanner messages. Each icon uses a different class, so there's `messageIconError`, `messageIconSuccess`, and `messageIconCancel`. I sent you the links to the Material Icons that we can use for Success and Cancel. Let me know if you need anything else!

<img width="349" alt="screen shot 2018-11-14 at 4 15 11 pm" src="https://user-images.githubusercontent.com/14582709/48516351-c4081500-e828-11e8-8df1-5a0fcca11e68.png">

**Comment by chas-mcmahon** [comment]
Got a few updates about the state of ticket scans

1) The basic Success/Already Redeemed/Error messages are all hooked up to the api and displaying the icons and copy from the mocks. The icons are the closest approximations I could find, but they should be good enough for now. I'm not going to have time to integrate the animation.

2) I commented out the guest info stuff at the bottom of the mocks because it's not ready. On the client side, some of the mocks haven't been implemented yet, eg [the guest info card represented here](https://app.goabstract.com/projects/b52ae8a0-8a13-11e8-a66c-3fc17367628a/branches/master/commits/fe40e1ee2343d01ac7934cd79b9dfeb2b51f9f64/files/AE300282-67A5-42F5-A47D-180F88B20490/layers/7CB1372A-9500-421B-BCE6-867DA61B4FDB). On the api side, there's no guest data sent back from the `redeem` request, so we're blocked on implementing guest info until we come up with a workable solution there. Ideally, I'd like to get all the relevant guest info back in one response so that we don't introduce latency from having to make multiple api calls per check-in, but we'll have to talk it out with Keith.

3) As far as I can tell there's no validation of the ticket against the event the doorman is currently scanning for, since the only data sent to the `redeem` action are the `ticket_id` and `redeem_key` from the QR scan. I talked about it with @Mcscott142 for a bit and it sounds like this doesn't matter for the pilot event coming up since that's the only event currently planned, but down the road the implication is that any bigneon ticket is more or less valid for any bigneon event. Another thing we'll have to collaborate with the api team on.

At this point I'm just waiting on review feedback on the implementation, but I think I'll be abl to wrap that up pretty quickly and move this to QA soon.

**Comment by zmelamed** [comment]
@krakaw 2) Associating a profile with a ticket when it's scanned--when/how will the client retrieve the guest info to be displayed related to the ticket scan? Will/should the device already have all the info it needs locally (per the requirement of being able to view a list of guests and their associated tickets for manual checkin)? We may need to sequence diagram the various flows relative to API calls 
3) a ticket needs to not just be redeemed as valid, but valid in the context of that specific event. Is that how this is setup to work?

**Comment by Mcscott142** [comment]
Meant to drop an update here yesterday. We need live and past events to QA this but having trouble creating Events on big neon staging.

**Comment by Mcscott142** [comment]
A fix was pushed so we could create events to test but I'm still having some issues. I can save a Draft event but I'm unable to publish an event with a relatively close start time to see if the Scanning is working on valid, live events.

**Comment by Krakaw** [comment]
I have prioritized big-neon/bn-web#420 for first thing today, hopefully that will resolve these issues.

**Comment by Krakaw** [comment]
> 2) Associating a profile with a ticket when it's scanned--when/how will the client retrieve the guest info to be displayed related to the ticket scan? 

Requires big-neon/bn-api#581

> Will/should the device already have all the info it needs locally (per the requirement of being able to view a list of guests and their associated tickets for manual checkin)? 

I think we should cache https://big-neon.github.io/bn-api-node/classes/_resources_namespaced_event_guests_.eventguestsresource.html#index on the device in case of internet connectivity issues.

> 3) a ticket needs to not just be redeemed as valid, but valid in the context of that specific event. Is that how this is setup to work?

We might need to change the endpoint to take an event_id as well big-neon/bn-api#581

**Comment by Mcscott142** [comment]
@Krakaw I can create an Event but now I'm having issues purchasing a ticket to be scanned. I can select a Qty but the purchasing of the tickets is unresponsive. See below. 

![2018-11-27 08 31 23](https://user-images.githubusercontent.com/6877299/49088475-f9483600-f21e-11e8-9753-21f9a417ebda.gif)

**Comment by Krakaw** [comment]
@Mcscott142 It's a bug with 0 value items not adding to the cart big-neon/bn-api#606, you can use https://staging.bigneon.com/events/4c6e824d-566f-42e5-9da0-692e153205a6 instead.

**Comment by jwietelmann** [comment]
@Mcscott142 you're hypothetically unblocked by this. ^ I'm removing the blocked label.

**Comment by Mcscott142** [comment]
Maybe the Event in that link ^ has passed or has been removed, sorry I'm just now getting to it. 

I'm still having difficulty testing the Scanning. I can only access QR codes for Events that have not started so I'm unable to test the successful and previously scanned states. 

I can now create an event in the Web App but I'm unable to purchase tickets for it as stated above. The last comment says it's been fixed https://github.com/big-neon/bn-api/issues/606#issuecomment-442371279, but I'm unable to purchase a ticket, same error happens. Maybe it's just not deployed? 

I can purchase tickets to other events but there is nothing with a start time sooner than Dec 6th.  I am signed in using the `superadmin` login but those already existing events that are available for purchase do not show in my Admin dashboard so I can edit the door times. 


Steps I am trying to take. 

1) Create an Event with a Doors open time happening currently
2) Purchase a ticket to that event - unable to do this. 
3) Scan that ticket to test for Valid and Already Redeemed states

Re-Marking as blocked. cc @Krakaw

**Comment by Krakaw** [comment]
@Mcscott142 There is a bug on the frontend with 0 value items big-neon/bn-web#451 if you give the event a price it will work. There is also currently no limit on how early you can scan the tickets (this is currently a bug in bn-api big-neon/bn-api#614)

**Comment by Mcscott142** [comment]
Ok, able to edit some events with start times. I don't think the Scanning is functioning properly. The only message I am able to get is `We Found an Issue` when Scanning a QR code. Moving this back to have someone check it out. 

Reproduction Steps:
1) Logged into [BN Staging Web App](https://staging.bigneon.com)  with an account created via the App `colin@revelry.co`
2)  Purchased Ticket to `Fancy Pants Party` and `It's My Party` (Both changed to have current door time and today's date)
3) Navigated to Fan Hub (under account name in Nav)
4) Select View my tickets on event
5) Select View QR Code
6) Open Doorman via the App
7) Select Event related to QR code
8) Scan QR Code
9) Always get the `We Found an Issue` Message.

**Comment by Krakaw** [comment]
@Mcscott142 it is a permission issue, try logging into the scanner app using `superuser@test.com` and `password` I'll look into why a normal user can't scan.

**Comment by Mcscott142** [comment]
Ah, yes, Thank You! @Krakaw. I got the valid response as superuser.

**Comment by Mcscott142** [comment]
Moving to QA, you must be signed in as superuser@test like the above comment suggests but the Scanning is functioning for Valid tickets.
